api.go

```diff
@@ -19,18 +19,36 @@ type api struct {
 	// value returns a pointer to a newly allocated Go variable able to
 	// represent the JSON object returned by the query.
 	value func() interface{}
+
+	// args is a struct that that the JSON request body is decoded into.
+	args struct {
+		ConferenceID int `json:"conference_id" db:"conference_id"`
+	}
 }
 
+// TODO: return an empty array instead of nothing if no results returned from database?
+
+// TODO: if args are required but not supplied, return a specific error message?
+
 func (a *api) serve(s *server) {
-	// TODO(mdempsky): Add mechanism for queries to specify custom
-	// arguments picked out of the request.
+	err := json.NewDecoder(s.r.Body).Decode(&a.args)
+	if err != nil && err != io.EOF {
```

Comment:
What's the `err != io.EOF` case intended to handle? I guess because some APIs don't take arguments?

If so, I'd suggest avoiding calling Decode in cases where it's expected to error. E.g., leave `a.args` as `nil` for APIs that don't take arguments, and `args.serve` can detect that and skip trying to decode in that case.

Reply:
Yes. I'll do that instead.